modules/KoreBuild.Tasks/CodeSign.targets

```diff
@@ -27,37 +38,36 @@
       <Output TaskParameter="MSBuildx86Path" PropertyName="MSBuildx86Path" />
     </GetPathToFullMSBuild>
 
-    <GenerateSignToolDataFile Condition=" '$(GenerateSignToolDataFile)' != 'false' "
-      Files="@(FilesToSign)"
-      Exclusions="@(FilesToExcludeFromSigning)"
-      OutputPath="$(_IntermediateSignToolDataFile)" />
-
     <ItemGroup>
-      <SignToolOptions Condition=" '$(MSBuildx86Path)' != '' " Include="-msbuildPath;$(MSBuildx86Path)" />
-      <SignToolOptions Condition=" '$(SignToolDryRun)' == 'true' "  Include="-test" />
-      <SignToolOptions Condition=" '$(SignType)' == 'test' " Include="-testSign" />
-      <SignToolOptions Include="-nugetPackagesPath" />
-      <SignToolOptions Include="$(NuGetPackageRoot)" />
-      <SignToolOptions Include="-intermediateOutputPath" />
-      <SignToolOptions Include="$(IntermediateDir)" />
-      <SignToolOptions Include="-config" />
-      <SignToolOptions Include="$(SignToolDataFile)" />
-      <SignToolOptions Include="$(SignToolDataWorkingDir)" />
-    </ItemGroup>
+      <!-- Reset internal item groups. -->
+      <_FileSignInfo Remove="@(_FileSignInfo)" />
+      <_ItemsToSign Remove="@(_ItemsToSign)" />
 
-    <PropertyGroup>
-      <_SignToolMessage>Running SignTool</_SignToolMessage>
-      <_SignToolMessage Condition=" '$(SignToolDryRun)' == 'true' ">$(_SignToolMessage) (dry-run)</_SignToolMessage>
-      <_SignToolOutputImportance>Normal</_SignToolOutputImportance>
-      <_SignToolOutputImportance Condition="'$(CI)' == 'true'">High</_SignToolOutputImportance>
-    </PropertyGroup>
+      <!--
+        Map KoreBuild items into SignTool task items.
+      -->
+      <_FileSignInfo Include="%(FilesToSign.FileName)%(FilesToSign.Extension)" CertificateName="%(FilesToSign.Certificate)" />
+      <_FileSignInfo Include="%(FilesToExcludeFromSigning.FileName)%(FilesToExcludeFromSigning.Extension)" CertificateName="None" />
 
-    <Message Importance="High" Text="$(_SignToolMessage)" />
+      <!--
+        Only pass in top-level items. FilesToSign contains items which will be nested in a .nupkg or .vsix.
+        If this list isn't filtered, SignTool task will overwrite files in the obj/ folders of projects which breaks
+        incremental compliation.
+      -->
+      <_ItemsToSign Include="%(FilesToSign.Identity)" Condition=" '%(FilesToSign.Container)' == '' " />
+    </ItemGroup>
 
-    <Run Condition=" '$(SignToolDataFile)' != '' "
-      FileName="$(RoslynSignToolPath)"
-      Arguments="@(SignToolOptions)"
-      StandardOutputImportance="$(_SignToolOutputImportance)" />
+    <Microsoft.DotNet.SignTool.SignToolTask
+      DryRun="$(SignToolDryRun)"
+      TestSign="$(SignToolTestSign)"
+      ItemsToSign="@(_ItemsToSign)"
+      FileSignInfo="@(_FileSignInfo)"
+      StrongNameSignInfo=""
```

Comment:
This is empty because we already strong-name signed everything with the private key as part of compilation, right?

Reply:
This API name was confusing to me too. It is not actually related to the _action_ of applying strong name signing. Rather, it's mean to map .NET assemblies with a given strong name to a code signing certificate.

Empty because the MSBuild task has marked this as `[Required]`, but we don't currently use this aspect of the sign tool's cert-picking logic.

Reply:
O I C. I think this made more sense after I read the other arcade signing issues in terms of how it filters/decides the list of things to do. I agree it's bizarre.

Reply:
I'm also removing the `[Required]` part of this parameter. I think it got added in the beginning but now it doesn't seem to make sense to have it required anymore.